### ISSUE TITLE
Kind information should be included in HTML docs

### ISSUE THREAD

**Issue by hdgarrood** [purescript/purescript#3866]
For example, consider the following module:

```purescript
module Main where

data PolyProxy a = PolyProxy

data ProxyType1 (a :: Type) = ProxyType1

data ProxyType2 :: Type -> Type
data ProxyType2 a = ProxyType2

class C1 a

class C2 :: forall k. k -> Constraint
class C2 a

class C3 :: Type -> Constraint
class C3 a
```
Here's how HTML docs for this module look right now:
https://gistpreview.github.io/?5449369e99533846c1b29d479305db14

`ProxyType1` and `ProxyType2` are the same, and `C1` and `C2` are the same. However, `PolyProxy` is different in quite an important way from both of the other proxy types defined here, and `C3` is different from the other classes, but the differences aren't reflected in the HTML docs.

One way of addressing this would be to include the kind signature in the docs on the line above the declaration for types and classes, using the inferred kind signature if none was provided, like we do for value declarations. However, this is redundant if individual type arguments have been annotated with kinds, like in `ProxyType1`. I'm not sure what the best approach is:

- Only render standalone kind signatures; strip kind annotations for individual arguments if present?
- Render kind annotations on each argument, and don't render kind signatures?
- Always render both standalone kind signatures and kind annotations on each argument?
- Other options?

**Comment by hdgarrood** [comment]
I feel like for data types or classes with lots of parameters and/or different kinds, it's more helpful to pair up the type parameters with their kinds. Consider Halogen’s Component, for instance: I’d definitely want annotations on each type parameter individually there. Having a stand-alone kind signature is also helpful in some cases though: “this class takes a parameter of kind `Type -> Type`, and, oh, lucky me, `Array` has that kind”. Mentally converting either way feels a bit tedious, so perhaps we should always render both.

**Comment by natefaubion** [comment]
One option may be to always show it as written, but provide alt tooltips for variables.
```
data Proxy :: forall k. k -> Type
data Proxy a  = Proxy
```
Where `a` has the tooltip `a :: k`

**Comment by garyb** [comment]
I like that, although it may not be very discoverable. I guess the dashed-underline definition hint thing might work.

**Comment by hdgarrood** [comment]
I’m generally not a huge fan of using tooltips, because you can only see one thing at a time, and they’re a bit awkward to use on mobile or if you prefer to use a keyboard more than a mouse, so I think I’d prefer annotating individual parameters over tooltips.

I think I’m leaning towards showing both; another reason for this is that you don’t know what the kinds of things are once all of their arguments have been applied if you only provide annotations for the parameters, so the standalone signature gives you more information.

We don’t show things exactly as written in quite a few places in docs already, for instance, data constructors are rendered as separate items in a <ul> underneath, and we reproduce instance lists directly under the types and classes they describe (without names) regardless of where they occur. I think it would be good for data declarations to look fairly consistent in HTML docs, even if they don’t necessary look as consistent in source files.

**Comment by garyb** [comment]
I think Nate's suggestion is showing both: the annotated kind, but then the tooltip is there to show the kind on each parameter.

**Comment by hdgarrood** [comment]
Right, I understand that all the information is present, but I think it could be irritating to have to hover over the type parameters to get the information you want out of them, versus being able to just see it all immediately.

**Comment by hdgarrood** [comment]
I'll remove this from v0.14.0 to avoid dragging it out too far.

**Comment by JordanMartinez** [comment]
I believe this was closed in #4100

**Comment by hdgarrood** [comment]
Oops, yeah, looks like this is a dupe of #3973
